IllegalStateException on reload with springloaded-1.1.5 + spring-4.0.0

trying to reload a spring web app (core/mvc/security, classic root- and web-contexts), i get the exception/stacktrace below when reloading any class.

i am not aware of any uninitialized/closed contexts or beanfactories (after all my app is up and running at that point and continues to do so despite the exception...)

env: jdk6, springloaded-1.1.5.RELEASE, spring-4.0.0.RELEASE

```
java.lang.reflect.InvocationTargetException
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.springsource.loaded.agent.SpringPlugin.reinvokeInitHandlerMethods(SpringPlugin.java:213)
    at org.springsource.loaded.agent.SpringPlugin.reloadEvent(SpringPlugin.java:120)
    at org.springsource.loaded.TypeRegistry.fireReloadEvent(TypeRegistry.java:1664)
    at org.springsource.loaded.ReloadableType.loadNewVersion(ReloadableType.java:397)
    at org.springsource.loaded.ReloadableType.reloadProxiesIfNecessary(ReloadableType.java:476)
    at org.springsource.loaded.ReloadableType.loadNewVersion(ReloadableType.java:399)
    at org.springsource.loaded.TypeRegistry.loadNewVersion(TypeRegistry.java:828)
    at org.springsource.loaded.agent.ReloadableFileChangeListener.fileChanged(ReloadableFileChangeListener.java:51)
    at org.springsource.loaded.agent.Watcher.determineChangesSince(FileSystemWatcher.java:235)
    at org.springsource.loaded.agent.Watcher.run(FileSystemWatcher.java:219)
    at java.lang.Thread.run(Thread.java:662)
Caused by: java.lang.IllegalStateException: BeanFactory not initialized or already closed - call 'refresh' before accessing beans via the ApplicationContext
    at org.springframework.context.support.AbstractRefreshableApplicationContext.getBeanFactory(AbstractRefreshableApplicationContext.java:170)
    at org.springframework.context.support.AbstractApplicationContext.getBeanNamesForType(AbstractApplicationContext.java:1051)
    at org.springframework.web.servlet.handler.AbstractHandlerMethodMapping.initHandlerMethods(AbstractHandlerMethodMapping.java:105)
    ... 15 more
```

--- a/springloaded/src/main/java/org/springsource/loaded/agent/SpringPlugin.java
+++ b/springloaded/src/main/java/org/springsource/loaded/agent/SpringPlugin.java
@@ -225,7 +225,9 @@ public class SpringPlugin implements LoadtimeInstrumentationPlugin, ReloadEventP
 							.println("SpringPlugin: clearing CachedIntrospectionResults for " + clazz.getName() + " removed=" + o);
 				}
 			} catch (Exception e) {
-				e.printStackTrace();
+				if (GlobalConfiguration.debugplugins) {
+					e.printStackTrace();
+				}
 			}
 		}
 	}

